Report score command progress

DJKero opened the pull request:
Closes #15 
- Interactive command to report the score with opponent, wins and loses fields.
- Handler checks for validity of the report by checking if the users involved are in the Thread.
- Report message is very basic.

We'll need to add a check to avoid accepting score reports with the bot on the opponent field.
We'll need to improve the report message and the followup embed for the opponent to accept or dispute the score.
We'll also need to handle the accept and dispute events from the embed and persist the score report on the database.

tristan-zander commented:
See my PR here: https://github.com/tristan-zander/runback/pull/15#pullrequestreview-1190939257

DJKero commented:
> See my PR here: [#15 (review)](https://github.com/tristan-zander/runback/pull/15#pullrequestreview-1190939257)

I think I've addressed all of the comments in the review, you can check on #15
If I missed something just review it and I'll make the changes needed.

tristan-zander commented on discord-bot/src/interactions/application_commands/matchmaking.rs:
```diff
@@ -230,6 +259,143 @@ impl InteractionHandler for MatchmakingCommandHandler {
+                    .http_client
+                    .interaction(self.utils.application_id)
+                    .create_followup(data.interaction.token.as_str())
+                    .content(format!("Sent a score report in <#{}>", channel).as_str())?
```
This message is a bit misleading because no score report was sent in the default matchmaking channel. I would prefer this line be removed until that logic is implemented.
```suggestion
```

tristan-zander commented on discord-bot/src/interactions/application_commands/matchmaking.rs:
```diff
@@ -230,6 +259,143 @@ impl InteractionHandler for MatchmakingCommandHandler {
+                    .content(format!("Sent a score report in <#{}>", channel).as_str())?
+                    .embeds(&[EmbedBuilder::new()
+                        .title("New score report")
+                        .description(format!("{:?} - {:?}", wins, loses))
```
Can you add the names of the players to this alongside their score? Discord embeds offer key-value fields that you can use to achieve this. https://docs.rs/twilight-util/0.14.0/twilight_util/builder/embed/struct.EmbedBuilder.html#examples

tristan-zander commented on discord-bot/src/interactions/application_commands/matchmaking.rs:
```diff
@@ -230,6 +259,143 @@ impl InteractionHandler for MatchmakingCommandHandler {
+                        .description(format!("{:?} - {:?}", wins, loses))
+                        .validate()?
+                        .build()])?
+                    .flags(MessageFlags::EPHEMERAL)
```
This ephemeral message flag doesn't seem to be respected right now. Not totally sure why, but it is probably best to remove until we figure it out.
Additionally, both users should be able to see this result, in my opinion. 
```suggestion
```

tristan-zander commented on discord-bot/src/interactions/application_commands/matchmaking.rs:
```diff
@@ -230,6 +259,143 @@ impl InteractionHandler for MatchmakingCommandHandler {
+                    .create_followup(data.interaction.token.as_str())
+                    .content(format!("Sent a score report in <#{}>", channel).as_str())?
+                    .embeds(&[EmbedBuilder::new()
+                        .title("New score report")
```
Can you make this title a bit more insightful? Something along the lines as "User Discord Nickname Reported Match Scores"

tristan-zander commented on discord-bot/src/interactions/application_commands/matchmaking.rs:
```diff
@@ -230,6 +259,143 @@ impl InteractionHandler for MatchmakingCommandHandler {
+                }
+
+                let wins_option_value = options
+                    .get(1)
```
I tried to re-order the values of the command, and it appears that it's always sent in the correct order. I don't foresee indexing the options like this to be an issue as I previously mentioned.

tristan-zander reviewed (changes requested):
All the code looks good! I have a few issues with the information that the embed returns, but it's nothing much.

DJKero commented on discord-bot/src/interactions/application_commands/matchmaking.rs:
```diff
@@ -230,6 +259,143 @@ impl InteractionHandler for MatchmakingCommandHandler {
+                let guild_settings = self
+                    .utils
+                    .get_guild_settings(
+                        data.command
+                            .guild_id
+                            .ok_or_else(|| anyhow!("command cannot be used in a DM"))?,
```
So it should look like this right?:
`let guild_settings = self
                    .utils
                    .get_guild_settings(data.guild_id)
                    .await?;`

tristan-zander replied:
Yup, that seems right!

tristan-zander reviewed (approved):
Looks good!
